aexpect/client.py

```diff
@@ -525,8 +525,9 @@ def __init__(self, command=None, a_id=None, auto_close=False, echo=False,
 
         # Start the thread in the background
         self.tail_thread = None
-        if termination_func or output_func:
-            self._start_thread()
+        if self.is_alive():
+            if termination_func or output_func:
+                self._start_thread()
```

Comment:
Nice catch, even though the thread dies immediately it's better not to run it in the first place.

Reply:
Would you please include a selftest?

Reply:
Actually without modifying the sources it'd be hard to see whether the `_start_thread` had been called or not, so unless you have an idea how to cleanly do that let me just add card to add `pickle` tests...: https://trello.com/c/GODRyeHr/1385-aexpect-add-pickle-selftests

Reply:
I'd need to take a closer look at the code to figure out `pickle` tests. Let's leave that on your side for now. If I come up with something before someone starts on that card I'll get in touch.